modcc/module.cpp

```diff
@@ -400,6 +419,23 @@ bool Module::semantic() {
         return false;
     }
 
+    if (has_symbol("net_receive", symbolKind::procedure)) {
+        auto net_rec_api = make_empty_api_method("net_rec_api", "net_receive");
+        if (net_rec_api.second) {
+            for (auto &s: net_rec_api.second->body()->statements()) {
+                if (s->is_assignment()) {
+                    for (const auto &id: state_vars) {
+                        auto coef = symbolic_pdiff(s->is_assignment()->rhs(), id);
+                        if(coef->is_number()) {
+                            linear &= (coef->is_number()->value() == 1 || coef->is_number()->value() == 0);
+                        }
```

Comment:
I think the test here has to be:
* if the coef is not a number, it is not linear.
* otherwise the coef has to have value exactly 1 if the left hand side is id, else 0.

Mathematically, if the net receive action assigns _f(s, w)_ to _s_, where _s_ is the vector of state variables, then the linearity requirement is _f(r+s, w)_ = _f(r, w)_ + _s_. This comes down to _f(s, w)_ = _s_ + _h(w)_ for some function _h_.